### ISSUE TITLE
Could not compile: cant find crate rust

### ISSUE THREAD

**Issue by fwsGonzo** [rust-fuzz/cargo-fuzz#169]
I've used libfuzzer a lot in C++, and decided to try it in Rust as my first project.
This is probably a beginner issue, but I managed to switch to nightly, call cargo init fuzz, and when running the default `fuzz_target_1` I get this:

```
 --> fuzz_targets/fuzz_target_1.rs:3:1
  |
3 | extern crate rust;
  | ^^^^^^^^^^^^^^^^^^ can't find crate
```
Clearly, some crate is missing, but why? And cargo install rust had no effect.

**Comment by Manishearth** [comment]
Is your crate named `rust`? Usually the crate there is the crate you're running cargo-fuzz within

**Comment by fwsGonzo** [comment]
My crate? I didn't consider that, and yes you are completely right. My project is in a folder called rust. I guess that makes it a crate in the world of Rust. What exactly is the error about?

**Comment by Manishearth** [comment]
You have to run `cargo fuzz init` in the top-level directory of a rust crate

**Comment by fwsGonzo** [comment]
Yes, I have done that, and it does show the targets when listed:
```
$ cargo fuzz list
fuzz_target_1
```

**Comment by ruuda** [comment]
I ran into the same issue. In my case the problem was that the parent project was a binary. It needs to be a library instead.

**Comment by your-diary** [comment]
Essentially the same as @ruuda's comment, but more concretely:

- Add `src/lib.rs` in addition to `src/main.rs` (as implied by #146).

- Or use `cargo new --lib` instead of `cargo new` when creating a project.
